### ISSUE TITLE
System.BadImageFormatException: Invalid compressed integer.

### ISSUE THREAD

**Issue by greenozon** [icsharpcode/ILSpy#1529]
ILSpy version 5.0.0.4801-preview2

```
System.BadImageFormatException: Invalid compressed integer.
   at System.Reflection.Throw.InvalidCompressedInteger()
   at System.Reflection.Metadata.BlobReader.ReadCompressedInteger()
   at System.Reflection.Metadata.Ecma335.SignatureDecoder`2.DecodeType(BlobReader& blobReader, Boolean allowTypeSpecifications, Int32 typeCode)
   at System.Reflection.Metadata.Ecma335.SignatureDecoder`2.DecodeType(BlobReader& blobReader, Boolean allowTypeSpecifications)
   at System.Reflection.Metadata.Ecma335.SignatureDecoder`2.DecodeMethodSignature(BlobReader& blobReader)
   at System.Reflection.Metadata.MethodDefinition.DecodeSignature[TType,TGenericContext](ISignatureTypeProvider`2 provider, TGenericContext genericContext)
   at ICSharpCode.Decompiler.TypeSystem.Implementation.MetadataMethod.DecodeSignature()
   at ICSharpCode.Decompiler.TypeSystem.Implementation.MetadataMethod.get_ReturnType()
   at ICSharpCode.Decompiler.CSharp.RequiredNamespaceCollector.CollectNamespaces(IEntity entity, MetadataModule module, CodeMappingInfo mappingInfo)
   at ICSharpCode.Decompiler.CSharp.RequiredNamespaceCollector.CollectNamespaces(IEntity entity, MetadataModule module, CodeMappingInfo mappingInfo)
   at ICSharpCode.Decompiler.CSharp.RequiredNamespaceCollector.CollectNamespaces(IEntity entity, MetadataModule module, CodeMappingInfo mappingInfo)
   at ICSharpCode.Decompiler.CSharp.RequiredNamespaceCollector.CollectNamespaces(IEntity entity, MetadataModule module, HashSet`1 namespaces)
   at ICSharpCode.Decompiler.CSharp.RequiredNamespaceCollector.CollectNamespaces(EntityHandle entity, MetadataModule module, HashSet`1 namespaces)
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DecompileTypes(IEnumerable`1 types)
   at ICSharpCode.Decompiler.CSharp.WholeProjectDecompiler.<>c__DisplayClass30_0.<WriteCodeFilesInProject>b__2(IGrouping`2 file)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass17_0`1.<ForWorker>b__1()
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object )

```

https://www.sendspace.com/file/viysnb

**Comment by siegfriedpammer** [comment]
The signature blob of an obfuscated method contains an invalid signature.

**Comment by greenozon** [comment]
yes, obfuscated, so the question is - is the issue related to rendering bad name (and asserting) or something else

here is how IL shows it (if I get it right - it gets stucked on empty names?)

![image](https://user-images.githubusercontent.com/765343/59151207-c84f8600-8a37-11e9-8419-6fafdd824199.png)


and here is how dnSpy handles it

![image](https://user-images.githubusercontent.com/765343/59151216-fb921500-8a37-11e9-9933-f41d715f0a6c.png)

to summarize - if it makes sense, lets make similar rendering approach?
just for the sake of browsability
